src/server/wesnothd/server.cpp

```diff
@@ -2813,33 +2773,41 @@ void server::searchlog_handler(const std::string& /*issuer_name*/,
 
 	*out << "IP/NICK LOG for '" << parameters << "'";
 
-	bool found_something = false;
-
 	// If this looks like an IP look up which nicks have been connected from it
 	// Otherwise look for the last IP the nick used to connect
 	const bool match_ip = (std::count(parameters.begin(), parameters.end(), '.') >= 1);
 
-	for(const auto& i : ip_log_) {
-		const std::string& username = i.nick;
-		const std::string& ip = i.ip;
-
-		if((match_ip && utils::wildcard_string_match(ip, parameters)) ||
-		  (!match_ip && utils::wildcard_string_match(utf8::lowercase(username), utf8::lowercase(parameters)))
-		) {
-			found_something = true;
-			auto player = player_connections_.get<name_t>().find(username);
+	if(!user_handler_) {
+		bool found_something = false;
+		
+		for(const auto& i : ip_log_) {
+			const std::string& username = i.nick;
+			const std::string& ip = i.ip;
+
+			if((match_ip && utils::wildcard_string_match(ip, parameters)) ||
+			(!match_ip && utils::wildcard_string_match(utf8::lowercase(username), utf8::lowercase(parameters)))
+			) {
+				found_something = true;
+				auto player = player_connections_.get<name_t>().find(username);
 
-			if(player != player_connections_.get<name_t>().end() && player->client_ip() == ip) {
-				*out << std::endl << player_status(*player);
-			} else {
-				*out << "\n'" << username << "' @ " << ip
-					 << " last seen: " << lg::get_timestamp(i.log_off, "%H:%M:%S %d.%m.%Y");
+				if(player != player_connections_.get<name_t>().end() && player->client_ip() == ip) {
+					*out << std::endl << player_status(*player);
+				} else {
+					*out << "\n'" << username << "' @ " << ip
+						<< " last seen: " << lg::get_timestamp(i.log_off, "%H:%M:%S %d.%m.%Y");
+				}
 			}
 		}
-	}
 
-	if(!found_something) {
-		*out << "\nNo match found.";
+		if(!found_something) {
+			*out << "\nNo match found.";
+		}
+	} else {
```

Comment:
I suppose there is some wildcard conversion needed? Our own wildcard matching works on `*` and `?` while SQL LIKE uses `%` and `_`. `_` is a bit problematic since it's allowed in user names. I would suggest to just convert `*` to `%` since it's the most useful anyway and escape `_`.